### PR TITLE
Update publish.py

### PR DESCRIPTION
**Pull request by yashbhedaru**
closes: #444 

Updated publish.py to take in a visualization object is needed which passes relevant visualization parameters through to the catalog file.

### CODE REVIEW

File: rubicon_ml/intake_rubicon/publish.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 def publish(
     experiments,
+    # visualization object passed, defaulted to None 
+    visualization_object=None,
```
Comment (stephenpardy) [inline rubicon_ml/intake_rubicon/publish.py:8]: We have been working to expand type hinting throughout the library, would request that we add mypy type hints to any parameters we are adding.

File: rubicon_ml/intake_rubicon/publish.py [hunk 2]
```diff
@@ -118,5 +122,24 @@ def _build_catalog(experiments):
 
         experiment_catalog_name = f"experiment_{experiment.id.replace('-', '_')}"
         catalog["sources"][experiment_catalog_name] = appended_experiment_catalog
+    
+    # create visualization entry to the catalog file
+    if visualization != None:
+        appended_visualization_catalog = {
+            "driver": "rubicon_ml_experiment_table",
+            "args": {
+                "experiments": visualization.experiments,
```
Comment (ryanSoley) [inline rubicon_ml/intake_rubicon/publish.py:131]: we won't need the experiments themselves. we'll assume they're present as actual experiment sources in the catalog too, and later add some checks to validate that

**Review by ryanSoley — CHANGES_REQUESTED** [review]
looking good! one small change request and the type hints like Stephen suggested

also, can you add a visualization object to this test?
> https://github.com/capitalone/rubicon-ml/blob/main/tests/unit/intake_rubicon/test_publish.py#L7

File: rubicon_ml/intake_rubicon/publish.py [hunk 3]
```diff
@@ -118,5 +122,24 @@ def _build_catalog(experiments):
 
         experiment_catalog_name = f"experiment_{experiment.id.replace('-', '_')}"
         catalog["sources"][experiment_catalog_name] = appended_experiment_catalog
+    
+    # create visualization entry to the catalog file
+    if visualization != None:
```
[the remark this answers is not in the document]
Comment (ryanSoley) [reply rubicon_ml/intake_rubicon/publish.py:127]: @yashbhedaru you can run `pre-commit install` to have the [style checks](https://capitalone.github.io/rubicon-ml/contributing.html#style) run on each commit

File: rubicon_ml/intake_rubicon/publish.py (hunk 1 above)
Comment (yashvb) [reply rubicon_ml/intake_rubicon/publish.py:8]: So would the code look something like this? " visualization_object: Optional[ExperimentsTable] = None" and I import the experiments table from the viz library into publish.py?

File: rubicon_ml/intake_rubicon/publish.py (hunk 2 above)
Comment (yashvb) [reply rubicon_ml/intake_rubicon/publish.py:131]: Okay I'll remove experiments from it.

File: rubicon_ml/intake_rubicon/publish.py (hunk 1 above)
Comment (ryanSoley) [reply rubicon_ml/intake_rubicon/publish.py:8]: yup! if you don't need the `ExperimentsTable` anywhere other than the type hint, you can guard the import behind a `TYPE_CHECKING` block
> https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

Comment (ryanSoley) [reply rubicon_ml/intake_rubicon/publish.py:8]: that way, it'll only actually get imported when you're type checking. not at runtime where its not actually needed

**Comment by ryanSoley** [comment]
closing in favor of #450

**Pull request closed without merging**